content/codespaces/getting-started-with-codespaces/getting-started-with-your-python-project-in-codespaces.md

```diff
@@ -0,0 +1,222 @@
+- **Settings** - These are {% data variables.product.prodname_vscode %} settings we wish to set
+  - **Terminal.integrated.shell.linux** - While bash is the default here, we could use zsh for example by modifying this.
+- **Extensions** - These are extensions included by default.
+  - **ms-python.python** - The Microsoft Python extension provides rich support for the Python language (for all actively supported versions of the language: >=3.6), including features such as IntelliSense, linting, debugging, code navigation, code formatting, refactoring, variable explorer, test explorer, and more.
+- **forwardPorts** - By default we can forward a port, like port 3000, but these will also forward automatically
+- **postCreateCommand** - If we want to run anything after we land in our codespace that’s not defined in our Dockerfile, like `pip3 install -r requirements`, we can do that here.
+- **remoteUser** - We’re running as the vscode user, but you can optionally set this to root
+
+##### Dockerfile
+
+```bash
+# [Choice] Python version: 3, 3.9, 3.8, 3.7, 3.6
+ARG VARIANT="3"
+FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+
+# [Option] Install Node.js
+ARG INSTALL_NODE="true"
+ARG NODE_VERSION="lts/*"
+RUN if [ "${INSTALL_NODE}" = "true" ]; then su vscode -c "umask 0002 && . /usr/local/share/nvm/nvm.sh && nvm install ${NODE_VERSION} 2>&1"; fi
+
+# [Optional] If your pip requirements rarely change, uncomment this section to add them to the image.
+# COPY requirements.txt /tmp/pip-tmp/
+# RUN pip3 --disable-pip-version-check --no-cache-dir install -r /tmp/pip-tmp/requirements.txt \
+#    && rm -rf /tmp/pip-tmp
+
+# [Optional] Uncomment this section to install additional OS packages.
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment this line to install global node packages.
+# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
+```
+
+You can use the Dockerfile to add additional container layers to specify OS packages, node versions, or global packages we want included in our Dockerfile.
+
+### Step 3: Modify your devcontainer.json file 
+
+With your dev container added and a basic understanding of what everything does, you can now make changes to configure it for your environment. In this example, you'll add properties to install npm when your codespace launches and make a list of ports inside the container available locally.
+
+1. In the Explorer, select the `devcontainer.json` file from the tree to open it. You might have to expand the `.devcontainer` folder to see it. 
+
+  !["Codespaces: Rebuild Container" in the command palette](/assets/images/help/codespaces/devcontainers-options.png)  
+
+2. Update your `devcontainer.json` file after `extensions` to include the following:
```

Comment:
I would love to suggest an extra extension to add for them that we'd recommend.

Reply:
Jupyter? Pylance? I believe this would also involve updating settings (i.e. setting Pylance as the language server).

Reply:
In our sample, we add the [Jinja Extension](https://marketplace.visualstudio.com/items?itemName=wholroyd.jinja) which provides highlighting for Jinja templates in the sample Flask project.

Reply:
Love this idea. Maybe the [spellchecker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) as well? [Flask snippets](https://marketplace.visualstudio.com/items?itemName=cstrap.flask-snippets)?